api/postgresql/db.go

```diff
@@ -168,7 +168,7 @@ func (service *Service) Save(query ...interface{}) error {
 		err := service.Insert(q)
 		if err != nil {
 			// oh dear god, only try and update if it fails because of a duplicated primary key.
-			if strings.HasSuffix(err.Error(), `_pkey"`) {
+			if strings.HasPrefix(err.Error(), "ERROR #23505 duplicate key value violates unique constraint") && strings.HasSuffix(err.Error(), `_pkey"`) {
```

Comment:
It's hair splitting, but I was thinking of something more simple, like `strings.Contains(err.Error(), "duplicate key value violates unique constraint")`

Maybe we'll have time this year to revisit and use `ON CONFLICT DO UPDATE` which became available in PostgreSQL 9.5.

Reply:
I'm also hoping that after this push we'll have time to move the last two tables into simple storage instead of continuing to use the old db code.

Reply:
@macrael What 2 tables are that?

Reply:
Looking at my notes it's 3. accounts, transmissions, and basic auth memberships.

Reply:
I'm not sure those are good candidates for the document-oriented simple storage – let's discuss if you pick that work up.

Reply:
I wasn't clear, I want to continue using the tables the same way, but re-implement the db code to just use sqlx and live inside the storage package rather than keeping the old db code around.